graph_snyk_cve_sync.py

```diff
@@ -190,124 +226,186 @@ def get_affected_versions(self, rules, versions):
                                 affected.append(ver)
         return list(set(affected))
 
-    def get_affected_versions_npm(self, rules, versions):
-        """Get affected versions for npm."""
-        affected = []
-        for ver in versions:
-            for rule in rules:
-                if Version(ver) in rule:
-                    affected.append(ver)
-                    break
-        return list(set(affected))
-
     def get_semver_versions(self, versions):
         """Convert to semver version format."""
         semver = []
         for ver in versions:
             semver.append(api.to_semver(ver))
         return semver
 
-    def extract_data_from_feed(self, data):
+    def extract_data_from_feed(self):
         """Fetch all the required info from the feed."""
-        for eco in data:
-            if eco == "java":
-                self.parse_maven_data(data[eco])
-            elif eco == "js":
-                self.parse_npm_data(data[eco])
-            elif eco == "python":
-                self.parse_pypi_data(data[eco])
+        for eco in self.snyk_data:
+            if eco == "java" and self.parse_data_for_eco(eco):
```

Comment:
can be changed to day in [0,3]. Since values do not change it can be initialised as a list upfront. Can minimise that many method calls.

Reply:
method was specifically added, as it contains an over riding env variable. If its delta feed on mode, then no matter which day, the feed needs to run. Didnt want to repeat that logic everywhere hence added to the function.

Reply:
condition in the function changed to in ["0", "3"].